File: Alay/Sliding-Window/1456.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int vowelCount = 0, maxVowelCount = 0;
        int start = 0;
        for (int i = 0; i < s.size(); i++) {
            vowelCount += isVowel(s[i]); 
            if (i >= k-1) {
                maxVowelCount = std::max(vowelCount, maxVowelCount);
                vowelCount -= isVowel(s[start]);
                start++;
            }
        }

        return maxVowelCount;
    }

    bool isVowel(char c) {
        return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u');
    }
};

File: Aditya-Pratap/Sliding Window/1456.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        new="aeiou"
        count=0
        for i in range(k):
            if s[i] in new:
                count+=1
        max_count=count
        i,j=0,k
        while j<len(s):
            if s[j] in new:
                count+=1
            if s[i] in new:
                count-=1
            max_count = max(max_count, count)  
            j+=1
            i+=1
        return max_count